### DISCUSSION TITLE
How to specify docker image command?

### DISCUSSION THREAD

**Discussion by justinpincar** [community/community#25086] [Actions]
<p>Trying to migrate from GitLab CI, and not sure how to specify commands for docker images.</p>
<p>Afaik, the mysql image doesn’t have env variables for these command-line options.</p>
<p>gitlab-ci.yml</p>
<pre><code>services:
    - name: mysql:5.7
      command: ["mysqld", "--lower-case-table-names=1", "--character-set-server=utf8mb4", "--collation-server=utf8mb4_unicode_ci"]
    - redis:5.0-alpine
</code></pre>
<p>What is the corresponding github yml syntax?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi,</p>
<p>Many thanks for your feedback!  </p>
<p>Github action has similar services structure as gitlab CI, please kindly check <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idservices" rel="nofollow noopener">here</a>.</p>
<p>The options can be set in the command in ‘<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsrun" rel="nofollow noopener">jobs.&lt;job_id&gt;.steps.run</a>’, code sample as below:</p>
<pre><code>services:
      mysql:
        image: mysql:5.7
        env:
          MYSQL_ROOT_PASSWORD: password
    steps:
      - name: echo
        run: |
           docker run -d mysql:5.7 --lower-case-table-names=1 --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci
</code></pre>

**Comment by justinpincar** [comment]
<p>Thanks, didn’t realize that you could just do a <code>docker run -d</code> in the CI environment.</p>
<p>I read about the original feature on GitLab and they said they basically just overwrite the CMD directive in the Dockerfile, so as a workaround, I just created a two-line Dockerfile with that modification myself.</p>
<pre><code>➜ usurp-mysql git:(master) cat Dockerfile.mysql
FROM mysql:5.7
CMD ["mysqld", "--lower-case-table-names=1", "--character-set-server=utf8mb4", "--collation-server=utf8mb4_unicode_ci"]
</code></pre>

**Comment by mieszko4** [comment]
<p>Running <code>docker run -d ...</code> will create a new instance that will not be used, no?</p>
<p>I used your solution but I removed services config completely and put all options directly into <code>docker run -d ...</code></p>
